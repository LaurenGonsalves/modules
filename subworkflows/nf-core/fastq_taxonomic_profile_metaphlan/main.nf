include { METAPHLAN_MAKEDB      } from '../../../modules/nf-core/metaphlan/makedb/main'
include { METAPHLAN_METAPHLAN     } from '../../../modules/nf-core/metaphlan/metaphlan/main'
include { METAPHLAN_MERGEMETAPHLANTABLES     } from '../../../modules/nf-core/metaphlan/mergemetaphlantables/main'


workflow FASTQ_TAXONOMIC_PROFILE_METAPHLAN {

    take:
    ch_fastq

    main:

    ch_versions = Channel.empty()

    METAPHLAN_MAKEDB ( )
    ch_versions = ch_versions.mix(METAPHLAN_MAKEDB.out.versions.first())

    METAPHLAN_METAPHLAN ( ch_fastq, METAPHLAN_MAKEDB.out.db )
    ch_versions = ch_versions.mix(METAPHLAN_METAPHLAN.out.versions.first())

    METAPHLAN_MERGEMETAPHLANTABLES ( METAPHLAN_METAPHLAN.out.profile.map{ [ [id:'all_samples'], it[1] ] }.groupTuple( sort:true) )
    ch_versions = ch_versions.mix(METAPHLAN_MERGEMETAPHLANTABLES.out.versions.first())

    emit:
    merged_taxa      = METAPHLAN_MERGEMETAPHLANTABLES.out.txt


    versions = ch_versions
}

