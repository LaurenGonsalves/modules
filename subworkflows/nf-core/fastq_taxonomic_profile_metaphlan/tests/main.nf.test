nextflow_workflow {

    name "Test Subworkflow FASTQ_TAXONOMIC_PROFILE_METAPHLAN"
    script "../main.nf"
    workflow "FASTQ_TAXONOMIC_PROFILE_METAPHLAN"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_taxonomic_profile_metaphlan"
    tag "metaphlan/makedb"
    tag "metaphlan/metaphlan"
    tag "metaphlan3/mergemetaphlantables"


    test("sarscov2 - fastq - paired_end") {

        when {
            workflow {
                """
                input[0] = Channel.of (
                    [
                        [ id:'test1', single_end:false ], // meta map
                            [
                                file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                                file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                            ]
                    ],
                    [
                        [ id:'test2', single_end:false ], // meta map
                            [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test2_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test2_2.fastq.gz', checkIfExists: true)
                            ]
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
