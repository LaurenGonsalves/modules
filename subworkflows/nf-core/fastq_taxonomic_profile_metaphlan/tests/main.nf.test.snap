{
    "sarscov2 - fastq - paired_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,b418dea99c622fcc1ca5f8dfc3fe8f3c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,562b214e2888a0f15d5afa6e787baed1",
                    "versions.yml:md5,7a8a0a934b8782f4bcb0b2de929e30e2",
                    "versions.yml:md5,cb828df59a55c1e7f350823185cfa2ad"
                ],
                "merged_taxa": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.txt:md5,b418dea99c622fcc1ca5f8dfc3fe8f3c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,562b214e2888a0f15d5afa6e787baed1",
                    "versions.yml:md5,7a8a0a934b8782f4bcb0b2de929e30e2",
                    "versions.yml:md5,cb828df59a55c1e7f350823185cfa2ad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T20:55:44.740041632"
    }
}